libs/commons/JSON.ml

```diff
@@ -17,6 +16,25 @@ type t =
   | Null
 [@@deriving show, eq]
 
+(* polymorphic variant style, used in Yojson.Basic.t *)
+type yojson =
+  [ `Null
+  | `Bool of bool
+  | `Int of int
+  | `Float of float
+  | `String of string
+  | `Assoc of (string * yojson) list
+  | `List of yojson list ]
```

Comment:
Why are we redefining this instead of just doing `type yojson = Yojson.Basic.t`?

Reply:
We could. I just wanted to gather in one place the many different ways the OCaml community represent
JSON values, and by putting the type there we can clearly see the difference.

Reply:
I could also have done type ezjsonm = Yaml.value, but I didn't want to add a dependency to the yaml library in OCaml,
and the cool thing about polymorphic variants is that you don't need the dependency!